.github/workflows/python-app.yml

```diff
@@ -46,3 +46,8 @@ jobs:
         run: poetry install
       - name: Run app
         run: poetry run python src/main.py ${{ secrets.GITHUB_TOKEN }}
+      - name: Deploy to GH pages
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: gen
```

Comment:
Nope. Only deploy if it's a commit on `main`. You definitely don't want to be triggering deployments on every CI run.